lib/galaxy/jobs/runners/drmaauniva.py

```diff
@@ -0,0 +1,640 @@
+"""
+Kind of fail save job control via the DRMAA API.
+Tested for UNIVA grid engine
+"""
+import logging
+import os
+import re
+import signal
+import subprocess
+import time
+
+# from galaxy import model
+from galaxy.jobs.runners.drmaa import DRMAAJobRunner
+
+
+log = logging.getLogger( __name__ )
+
+__all__ = ( 'DRMAAUnivaJobRunner', )
+
+MEMORY_LIMIT_SCAN_SIZE = 1024 * 1024  # 1MB
+
+"""
+TODO
+find out what it means if there is no error code file (.ec)
+can this also be used
+
+tool wrapper:
+- A tool that is the bash script itself.
+- bash script (create long array, dd into variable)
+- C
+- C++
+- python
+- java
+- perl
+"""
+
+
+class DRMAAWaitUnusable( Exception ):
+    pass
+
+
+class DRMAAUnivaJobRunner( DRMAAJobRunner ):
+    runner_name = "DRMAAUnivaJobRunner"
+    # restrict job name length as in the DRMAAJobRunner
+    # restrict_job_name_length = 15
+
+    def check_watched_item( self, ajs, new_watched ):
+        # get state with job_info/qstat
+        state = self._get_drmaa_state(ajs.job_id, self.ds, False)
+        logging.debug("DRMAAUnivaJobRunner:check_watched_item ({jobid}) -> state {state}".format(jobid=ajs.job_id, state=self.drmaa_job_state_strings[state]))
```

Comment:
These log statements should use the log object acquired at the top, i.e. `logging.debug` -> `log.debug`. Also, for consistency, you shouldn't need to preface with `DRMAAUnivaJobRunner` since the module name is included in the message.

Reply:
has been replaced in the whole file